### PR TITLE
ENH Replace gpuci_conda_retry with gpuci_mamba_retry

### PR DESCRIPTION
**Pull request by dillon-cullinan**
`mamba` was recently added to gpuCI build environment, testing usage and solvability with this PR which should speed up build times.

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Dillon! 😄

This seems reasonable, but it looks like Jenkins is running into some kind of issue:

```
Error response from daemon: manifest for gpuci/rapidsai-driver:21.10-cuda11.0-devel-centos7-py3.7 not found: manifest unknown: manifest unknown
Docker pull failed, retry 1 of 3...
```

Is this know/being tracked somewhere?

**Comment by dillon-cullinan** [comment]
rerun tests

**Review by jakirkham — APPROVED** [review]

**Comment by quasiben** [comment]
@dillon-cullinan I think we can merge this after removing the `Draft` status

**Comment by jakirkham** [comment]
I touched base with Dillon offline and he said this is on hold for the moment

**Comment by dillon-cullinan** [comment]
Yes, there is a large batch of these PRs and we want to merge them at the same time. There are only two that are failing right now due to some missing 21.10 packages. Just waiting on those.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by dillon-cullinan** [comment]
@gpucibot merge

**Pull request merged**
